src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelProxyExecutionManager.cs

```diff
@@ -287,32 +287,15 @@ private void StartTestRunOnConcurrentManager(IProxyExecutionManager proxyExecuti
                 if (this.TryFetchNextSource(this.sourceEnumerator, out string nextSource))
                 {
                     EqtTrace.Info("ProxyParallelExecutionManager: Triggering test run for next source: {0}", nextSource);
-
-                    testRunCriteria = new TestRunCriteria(
-                                          new[] { nextSource },
-                                          this.actualTestRunCriteria.FrequencyOfRunStatsChangeEvent,
-                                          this.actualTestRunCriteria.KeepAlive,
-                                          this.actualTestRunCriteria.TestRunSettings,
-                                          this.actualTestRunCriteria.RunStatsChangeEventTimeout,
-                                          this.actualTestRunCriteria.TestHostLauncher)
-                                          {
-                                              TestCaseFilter = this.actualTestRunCriteria.TestCaseFilter
-                                          };
+                    testRunCriteria = new TestRunCriteria(new[] { nextSource }, this.actualTestRunCriteria);
```

Comment:
>actualTestRunCriteria [](start = 85, length = 21)

how was parallel ever working?

Reply:
When FilterOptions are used, it never works. otherwise with test case filter it does.